pyramid_swagger/tween20.py

```diff
@@ -0,0 +1,224 @@
+        """
+        self.request = request
+        self.route_info = route_info
+
+    @property
+    def query(self):
+        """
+        :rtype: dict
+        """
+        # The `mixed` dict will return a list if a parameter has multiple
+        # values or a single primitive in the case of a single value.
+        return self.request.GET.mixed()
+
+    @property
+    def form(self):
+        """
+        :rtype: dict
+        """
+        return self.request.POST.mixed()
+
+    @property
+    def files(self):
+        result = {}
+        for k, v in self.request.params.mixed():
+            if hasattr(v, 'file'):
+                result[k] = v.file
+        return result
+
+    @property
+    def path(self):
+        return self.route_info.get('match') or {}
+
+    @property
+    def headers(self):
+        return self.request.headers
+
+    def json(self, **kwargs):
+        return getattr(self.request, 'json_body', {})
+
+
+@validation_error(RequestValidationError)
+def swaggerize_request(request, settings, route_info):
+    """
+    Delegate handling the Swagger concerns of the request to bravado-core.
+    Post-invocation, the Swagger request parameters are available as a dict
+    named `swagger_data` on the Pyramid request.
+
+    :type request: :class:`pyramid.request.Request`
+    :type settings: :class:`Settings`
+    :type route_info: :class:`pyramid.urldispatch.Route`
+    """
+    op = get_op_for_request(request, route_info, settings.spec)
+    bravado_request = PyramidSwaggerRequest(request, route_info)
+    request_data = unmarshal_request(bravado_request, op)
+
+    def swagger_data(_):
+        return request_data
+
+    request.set_property(swagger_data)
+
+
+@validation_error(ResponseValidationError)
+def swaggerize_response(response):
+    """
+    Delegate handling the Swagger concerns of the response to bravado-core.
+
+    :type response: :class:`pyramid.response.Response`
+    """
+    # TODO: validate, marshal, and transform the response object
+    log.warn('TODO: Implement swaggerize_response()')
+
+
+def get_op_for_request(request, route_info, spec):
+    """
+    Find out which operation in the Swagger schema corresponds to the given
+    pyramid request.
+
+    :type request: :class:`pyramid.request.Request`
+    :type route_info: dict (usually has 'match' and 'route' keys)
+    :type spec: :class:`bravado_core.spec.Spec`
+    :rtype: :class:`bravado_core.operation.Operation`
+    :raises: RequestValidationError when a matching Swagger operation is not
+        found.
+    """
+    # pyramid.urldispath.Route
+    route = route_info['route']
+    if hasattr(route, 'path'):
+        op = spec.get_op_for_request(request.method, route.path)
+        if op is not None:
+            return op
+    raise PathNotMatchedError(
+        "Could not find a matching Swagger operation for {0} request {1}"
+        .format(request.method, request.url))
+
+
+def load_settings(registry):
```

Comment:
I think with the proposed changes to `Settings` you could also drop this, since it would be identical.